app/script/announce/AnnounceRepository.coffee

```diff
@@ -30,20 +32,26 @@ class z.announce.AnnounceRepository
 
   init: ->
     window.setTimeout =>
-      @fetch_announcements()
-      @schedule_check()
-    , CHECK_TIMEOUT
+      @check_announcements()
+      @schedule_checks()
+    , ANNOUNCE_CONFIG.CHECK_TIMEOUT
 
-  fetch_announcements: =>
+  check_announcements: =>
     @announce_service.get_announcements()
     .then @process_announce_list
     .catch (error) =>
       @logger.error "Failed to fetch announcements: #{error}"
 
+  check_version: =>
+    @announce_service.get_version()
+    .then (server_version) ->
+      amplify.publish z.event.WebApp.LIFECYCLE.UPDATE if server_version > z.util.Environment.version false, true
```

Comment:
This is confusing I have to admit.. Maybe something like 
```
needs_update = server_version > z.util.Environment.version...
```

and then user the variable?